src/Repository/PersistableEntityInterface.php

```diff
@@ -11,5 +11,5 @@ public function toElastic(): array;
      * @param array $document the raw document as found in the _source field of the raw Elasticsearch response
      * @param array $metaData contains all "underscore-fields" delivered in the raw Elasticsearch response (e.g. _score)
      */
-    public static function fromElasticDocument(array $document, array $metaData): object;
+    public static function fromElasticDocument(array $document, array $metaData): ?object;
```

Comment:
We have a use case on reviews service that depends on this. Basically it starts creating the document but then if some criteria is met it is returning null (and the the iterator result is filtered to remove "null" documents). On previous versions it was working because the return was not strongly enforced but rather a docblock @return object

Reply:
But I am ok with making it return object and try change the code there to another approach.

Reply:
i'd prefer that, yes.
my point is: if there is a document in ES, we should be able to create an object. and if there's no document, there's no point in calling the factory method (especially because $document strictly has to be an array)

Reply:
Ok, I will do that

Reply:
Done